src/sage/rings/finite_rings/finite_field_base.pyx

```diff
@@ -2120,6 +2120,89 @@ cdef class FiniteField(Field):
         python_int = int.from_bytes(input_bytes, byteorder=byteorder)
         return self.from_integer(python_int)
 
+    def _roots_univariate_polynomial(self, f, ring, multiplicities, **kwargs):
+        r"""
+        Return the roots of the univariate polynomial ``f``.
+
+        INPUT:
+
+        - ``f`` - a polynomial defined over this field
+
+        - ``ring`` - the ring to find roots in.
+
+        - ``multiplicities`` - bool (default: ``True``). If ``True``, return
+          list of pairs `(r, n)`, where `r` is a root and `n` is its
+          multiplicity. If ``False``, just return the unique roots, with no
+          information about multiplicities.
+
+        - ``kwargs`` - ignored
```

Comment:
doesn't look right. I suggest changing `**kwargs` to `algorithm=None`, then in the body, `if algorithm is not None: raise NotImplementedError`, which allows `algorithm=pari` to work automatically.